src/com/carolinarollergirls/scoreboard/game/TeamInfo.java

```diff
@@ -25,136 +25,42 @@ public String getUpdaterBase() {
 	}
 
 	public void snapshot() {
-		Team t = game.getTeam(team);
-		name = t.getName();
-		logo = t.getLogo();
-
-		// Add/Update/Remove Alternate Names
-		List<Team.AlternateName> t_an = t.getAlternateNames();
-		for (Team.AlternateName an : t_an) {
-			if (an != null && an.getId() != null && an.getName() != null)
-				alternateNames.put(an.getId(), an.getName());
-		}
-		Object[] keys = alternateNames.keySet().toArray();
-		for (Object oid : keys) {
-			String id = (String)oid;
-			boolean found = false;
-			for (Team.AlternateName an : t_an) {
-				if (an != null && id.equals(an.getId()) && an.getName() != null) {
-					found = true;
-					break;
-				}
-			}
-			if (!found) {
-				alternateNames.remove(id);
-				update("AlternateName(" + id + ")", null);
-			}
-		}
-
-		// Add/Update/Remove Colors
-		List<Team.Color> t_c = t.getColors();
-		for (Team.Color c : t_c) {
-			if (c != null && c.getId() != null && c.getColor() != null)
-				colors.put(c.getId(), c.getColor());
-		}
-		keys = colors.keySet().toArray();
-		for (Object oid : keys) {
-			String id = (String)oid;
-			boolean found = false;
-			for (Team.Color c : t_c) {
-				if (c != null && id.equals(c.getId()) && c.getColor() != null) {
-					found = true;
-					break;
-				}
-			}
-			if (!found) {
-				colors.remove(id);
-				update("Color(" + id + ")", null);
-			}
-		}
-
-		// Add/Update/Remove Skaters
-		List<Skater> t_s = t.getSkaters();
-		for (Skater s : t_s) {
-			boolean found = false;
-			for (SkaterInfo si : skaters) {
-				if (s != null && si.getId().equals(s.getId())) {
-					si.snapshot(s);
-					found = true;
-					break;
-				}
-			}
-			if (!found)
-				skaters.add(new SkaterInfo(game, this, s));
-		}
-		SkaterInfo[] sa = new SkaterInfo[skaters.size()];
-		skaters.toArray(sa);
-		for (SkaterInfo si : sa) {
-			boolean found = false;
-			for (Skater s : t_s) {
-				if (s != null && si.getId().equals(s.getId())) {
-					found = true;
-					break;
-				}
-			}
-			if (!found) {
-				skaters.remove(si);
-				update("Skater(" + si.getId() + ")", null);
-			}
-		}
-
-		queueUpdates();
 	}
 
 	public JSONObject toJSON() throws JSONException {
+		Team t = game.getTeam(team);
 		JSONObject json = new JSONObject();
 		json.put("team", team);
-		json.put("name", name);
-		json.put("logo", logo);
-		json.put("color", alternateNames.get("operator"));
+		json.put("name", t.getName());
+		json.put("logo", t.getLogo());
+		Team.AlternateName operatorName = t.getAlternateName("operator");
+		if (operatorName != null) {
+			json.put("color", operatorName.getName());
+		}
```

Comment:
Assuming that the operator name (if set) will be the team color is bound to break at some point. We should be looking to make this attribute accessible directly (and then add an operator option to use it as operator name, if set).
(I am aware that this is not in the scope of this PR.)

Reply:
This data is already available directly, and when I replace all of this this special casing will go away.